pkg/config/derive_test.go

```diff
@@ -0,0 +1,298 @@
+	os.Setenv("OREG_URL", initialOreg)
+}
+
+func Test_derived_ImageResourceName(t *testing.T) {
+	tests := []struct {
+		want string
+	}{
+		{
+			want: "test-image1",
+		},
+		{
+			want: "test-image2",
+		},
+	}
+	defer os.Setenv("IMAGE_RESOURCENAME", os.Getenv("IMAGE_RESOURCENAME"))
+
+	for _, tt := range tests {
+		t.Run(tt.want, func(t *testing.T) {
+			d := derived{}
+			os.Setenv("IMAGE_RESOURCENAME", tt.want)
+			if got := d.ImageResourceName(); got != tt.want {
+				t.Errorf("derived.ImageResourceName() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_derived_RunningUnderTest(t *testing.T) {
+	tests := []struct {
+		value string
+		want  bool
+	}{
+		{
+			value: "yes",
+			want:  true,
+		},
+		{
+			value: "anything",
+			want:  true,
+		},
+		{
+			value: "",
+			want:  false,
+		},
+	}
+	defer os.Setenv("RUNNING_UNDER_TEST", os.Getenv("RUNNING_UNDER_TEST"))
+
+	for _, tt := range tests {
+		t.Run(tt.value, func(t *testing.T) {
+			d := derived{}
+			os.Setenv("RUNNING_UNDER_TEST", tt.value)
+			if got := d.RunningUnderTest(); got != tt.want {
+				t.Errorf("derived.RunningUnderTest() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_derived_CloudProviderConf(t *testing.T) {
+	tests := []struct {
+		name    string
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "basic",
+			want: []byte(`aadClientId: client_id
+aadClientSecret: client_secrett
+location: eastus
+primaryScaleSetName: ss-compute
+resourceGroup: rg
+securityGroupName: nsg-compute
+subscriptionId: sub
+tenantId: tenant
+vmType: vmss
+`),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			d := derived{}
+			var cs = fixtures.NewTestOpenShiftCluster()
+			got, err := d.CloudProviderConf(cs)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("derived.CloudProviderConf() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("derived.CloudProviderConf() = \"%v\", want \"%v\"", string(got), string(tt.want))
+			}
+		})
+	}
+}
+
+func Test_derived_MasterLBCNamePrefix(t *testing.T) {
+	tests := []struct {
+		want string
+	}{
+		{
+			want: "example",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.want, func(t *testing.T) {
+			d := derived{}
+			var cs = fixtures.NewTestOpenShiftCluster()
+			if got := d.MasterLBCNamePrefix(cs); got != tt.want {
+				t.Errorf("derived.MasterLBCNamePrefix() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_derived_RouterLBCNamePrefix(t *testing.T) {
+	tests := []struct {
+		want string
+	}{
+		{
+			want: "router-fqdn",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.want, func(t *testing.T) {
+			d := derived{}
+			var cs = fixtures.NewTestOpenShiftCluster()
+			if got := d.RouterLBCNamePrefix(cs); got != tt.want {
+				t.Errorf("derived.RouterLBCNamePrefix() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_derived_PublicHostname(t *testing.T) {
+	tests := []struct {
```

Comment:
In all of these, I'm a little worried about the load we're putting on NewTestOpenShiftCluster.  Also this isn't really achieving full coverage of the function under test.  Wouldn't something like the following be better:

```
{
	in: OpenShiftManagedCluster{Properties: &Properties{PublicHostname: "foo", FQDN: "bar"}},
	want: "foo",
},
{
	in: OpenShiftManagedCluster{Properties: &Properties{FQDN: "bar"}},
	want: "bar",
},
```

(need to reconsider multiple test functions in this file accordingly)